### PR TITLE
Add links to documentation for v1.4 release note

### PR DESCRIPTION
**Pull request by emile-00**
## Info

- **Description**

  - Add links to documentation for v1.4 release note. @CharlieSYH @hengm3467 The file path format doesn't work in the preview. What format should we use for these links? 

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - [ Provide a link to the relevant code PR here, if applicable. ]

- **Related doc issue**
  
  Resolves [ Provide a link to the relevant doc issue here, if applicable. ]

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: src/pages/release-notes.md [hunk 1]
```diff
@@ -20,39 +20,39 @@ This version was released on November 10, 2023.
 - Query syntax:
   - Supports using subqueries in `UPDATE` and `DELETE` statements. [#12995](https://github.com/risingwavelabs/risingwave/pull/12995)
 - SQL commands
-  - Supports `COMMENT ON` clause for tables and columns. [#12849](https://github.com/risingwavelabs/risingwave/pull/12849)
-  - Supports persistent background materialized views. [#12167](https://github.com/risingwavelabs/risingwave/pull/12167)
+  - Supports `COMMENT ON` clause for tables and columns. [#12849](https://github.com/risingwavelabs/risingwave/pull/12849). [See documentation.](/versioned_docs/version-1.4/sql/commands/sql-comment-on.md)
```
Comment (hengm3467) [inline src/pages/release-notes.md]: @CharlieSYH Please take a look at the relative paths.

**Review by CharlieSYH — APPROVED** [review]
Updated all the doc links. For now, I'm using the "/current" path instead of "/1.4" to avoid redirection and ensure that readers can return to the release notes page easily.

File: src/pages/release-notes.md (hunk 1 above)
Comment (CharlieSYH) [reply src/pages/release-notes.md]: done

**Comment by hengm3467** [comment]
@emile-00 I found the applicable style recommendation for handling link text: https://developers.google.com/style/link-text

In summary, we should spell out the titles of the linked pages. Thanks.

**Pull request merged**